src/tools/PICS-generator/PICSGenerator.py

```diff
@@ -0,0 +1,429 @@
+import argparse
```

Comment:
Can you throw the licence header on here?

Reply:
@cecille  how this could be later improved to use the new dm-editor tool ?

Reply:
Of cause, my mistake. I will get this sorted.

Reply:
I need to get a deeper look at the DM XML, but I might be able to use these over the JSON to get PICS code, the cluster name is kind of already available in a cleaner way in python now, but given I still needed the PICS code, I just kept the using the name from that as well.

Reply:
I think we can actually just do this from the matter json dump and it's perfectly fine. The DM tools might factor in in that we probably want to ensure that all the stuff ON the device makes sense before generating a PICS. Ie, run IDM-10.1 and the conformance checker and the cluster revision checker before running the PICS tool generation, because none of those need PICS. Then we can just generate the PICS from the dump. The change here to do that would be fairly minimal - just swap out the read calls for a call to the dumped wildcard. But that's somewhat separate from the DM editor tools because the pics relate to what's on the device rather than that's in the spec. so as long as the device is conformant, the DM stuff has kind of just done its job and we can generate the pics here as rene has done and it's all good.

Meanwhile, if you pop a licence header on here, I'd argue that we should just get this in and we can have other folks hop in to do the refinements.